src/deepEqual.js

```diff
@@ -20,7 +20,28 @@
  * @return {boolean}
  */
 function deepEqual(a, b) {
-  // write code here
+  if (a === b) {
+    return true;
+  } else if (
+    typeof a === 'object'
+    && typeof a === 'object'
+    && a !== null
+    && b !== null
+  ) {
+    if (Object.keys(a).length !== Object.keys(b).length) {
+      return false;
+    }
+    for (const prop in a) {
+      if (a.hasOwnProperty(prop) !== b.hasOwnProperty(prop)) {
```

Comment:
Сейчас у нас нет наследования, тесты пройдут и без этого. Важно понимать что этот кусок кода делает и зачем он нужен.

Reply:
мы ведь ключи перебираем
```suggestion
      if (a.hasOwnProperty(key) !== b.hasOwnProperty(key)) {
```